src/Functions/seriesDecomposeAnomaliesDetection.cpp

```diff
@@ -0,0 +1,277 @@
+            //extract residual component of the decomposed series
+            ColumnArray::Offset residual_start_offset = nested_decompose_offsets[prev_decompose_offset + 1];
+            ColumnArray::Offset residual_end_offset = nested_decompose_offsets[prev_decompose_offset + 2];
+
+            auto residuals = ColumnFloat32::create();
+            auto & residuals_data = residuals->getData();
+            ColumnArray::ColumnOffsets::MutablePtr residuals_col_offsets = ColumnArray::ColumnOffsets::create();
+            auto & residuals_col_offsets_data = residuals_col_offsets->getData();
+
+            residuals_data.insertByOffsets(decompose_data, residual_start_offset, residual_end_offset);
+            residuals_col_offsets_data.push_back(residuals_data.size());
+
+            ColumnWithTypeAndName residuals_col;
+            residuals_col.type = std::make_shared<DataTypeArray>(std::make_shared<DataTypeNumber<Float32>>());
+            residuals_col.column = ColumnArray::create(std::move(residuals), std::move(residuals_col_offsets));
+
+            auto min_percentile_column = DataTypeFloat64().createColumnConst(1, min_percentile);
+            auto max_percentile_column = DataTypeFloat64().createColumnConst(1, max_percentile);
+            auto k_value_column = DataTypeFloat64().createColumnConst(1, K_value);
+
+            ColumnsWithTypeAndName outliers_arg{
+                std::move(residuals_col),
+                ColumnWithTypeAndName(min_percentile_column, std::make_shared<DataTypeFloat64>(), "min_percentile"),
+                ColumnWithTypeAndName(max_percentile_column, std::make_shared<DataTypeFloat64>(), "max_percentile"),
+                ColumnWithTypeAndName(k_value_column, std::make_shared<DataTypeFloat64>(), "K")};
+
+            auto get_outliers = FunctionFactory::instance().get("seriesOutliersDetectTukey", context)->build(outliers_arg);
+            auto get_outliers_res_type = std::make_shared<DataTypeArray>(std::make_shared<DataTypeFloat64>());
+            auto get_outliers_res = get_outliers->execute(outliers_arg, get_outliers_res_type, input_rows_count);
+
+            const IColumn & outlier_score_col = checkAndGetColumn<ColumnArray>(get_outliers_res.get())->getData();
+            const PaddedPODArray<Float64> & outlier_score = checkAndGetColumn<ColumnVector<Float64>>(&outlier_score_col)->getData();
+
+            std::transform(
+                outlier_score.begin(),
+                outlier_score.end(),
+                std::back_inserter(res_data),
+                [](Float64 score)
+                {
+                    if (score > 0.0)
+                        return 1;
+                    else if (score < 0.0)
+                        return -1;
+                    return 0;
+                });
+
+            res_col_offsets_data.push_back(res_data.size());
+
+            res_data.insert(outlier_score.begin(), outlier_score.end());
+            res_col_offsets_data.push_back(res_data.size()); //fix this
```

Comment:
why  `res_col_offsets_data.push_back(res_data.size());` twice?

Reply:
l.188 is to add offset for data inserted at l.75
l.191 is to add offset for data inserted at l.190